stan/math/prim/mat/fun/to_matrix.hpp

```diff
@@ -14,54 +14,53 @@ namespace stan {
      * @tparam T type of the scalar
      * @tparam R number of rows
      * @tparam C number of columns
-     * @param matrix matrix
+     * @param x matrix
      * @return the matrix representation of the input
      */
     template <typename T, int R, int C>
     inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
-    to_matrix(const Eigen::Matrix<T, R, C>& matrix) {
-      return matrix;
+    to_matrix(const Eigen::Matrix<T, R, C>& x) {
+      return x;
     }
 
     /**
      * Returns a matrix representation of the vector in column-major order
      * with the specified number of rows and columns.
      *
      * @tparam T type of the scalar
-     * @param vec vector of values
+     * @param x vector of values
      * @param m rows
      * @param n columns
      * @return the matrix representation of the input
      * @throw <code>std::invalid_argument</code> if the sizes do not match
      */
     template <typename T>
     inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
-    to_matrix(const std::vector<T>& vec, int m, int n) {
+    to_matrix(const std::vector<T>& x, int m, int n) {
       static const char* fun = "to_matrix(array)";
-      check_size_match(fun, "rows * columns", m * n, "vector size", vec.size());
+      check_size_match(fun, "rows * columns", m * n, "vector size", x.size());
       return Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic,
-                                            Eigen::Dynamic> >(&vec[0], m, n);
+                                            Eigen::Dynamic> >(&x[0], m, n);
     }
 
     /**
      * Returns a matrix representation of the standard vector of standard vectors
      * with the same dimensions and indexing order.
      *
      * @tparam T type of the scalar
-     * @param vec vector of vectors of scalar values
+     * @param x vector of vectors of scalar values
      * @return the matrix representation of the input
      */
     template <typename T>
     inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
-    to_matrix(const std::vector< std::vector<T> >& vec) {
-      size_t R = vec.size();
-      if (R != 0) {
-        size_t C = vec[0].size();
-        Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> result(R, C);
-        T* datap = result.data();
-        for (size_t i=0, ij=0; i < C; i++)
-          for (size_t j=0; j < R; j++, ij++)
-            datap[ij] = vec[j][i];
+    to_matrix(const std::vector< std::vector<T> >& x) {
+      size_t rows = x.size();
+      if (rows != 0) {
+        size_t cols = x[0].size();
+        Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> result(rows, cols);
+        for (size_t i=0, ij=0; i < cols; i++)
```

Comment:
Use `++i` rather than `i++` --- compiler can probably deal here, but only use `x++` when you are going to use the value of `x` before the increment.